src/components/VideoDetails/VideoDetails.component.jsx

```diff
@@ -1,29 +1,31 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useLocation, useParams } from 'react-router';
 import { Grid } from '@material-ui/core';
 import { getVideos, loadVideo } from '../../services/videoService';
-import SearchContext from '../../providers/SearchContext';
+import { useGlobal } from '../../providers/GlobalContext/GlobalContext';
 import RelatedCard from './RelatedCard';
 
-export default function VideoDetails({ loadVideoFn = loadVideo, getVideosFn = getVideos }) {
+export default function VideoDetails({
+  loadVideoFn = loadVideo,
+  getVideosFn = getVideos,
+}) {
   const [video, setVideo] = useState();
   const [relatedVideos, setRelatedVideos] = useState([]);
   const location = useLocation();
   const { videoId } = useParams();
-  const searchContext = useContext(SearchContext) || {};
+  const globalContext = useGlobal();
 
   useEffect(() => {
-
-    if (typeof location.video === undefined || location.video === null) {
+    if (typeof location.video === 'undefined' || location.video === null) {
       loadVideoFn(setVideo, videoId);
     } else {
       setVideo(location.video);
     }
-  }, [videoId, location.video]);
+  }, [videoId, location.video, loadVideoFn]);
 
   useEffect(() => {
-    getVideosFn(setRelatedVideos, searchContext.searchState);
-  }, [searchContext.searchState]);
+    getVideosFn(setRelatedVideos, globalContext.searchState);
+  }, [globalContext.searchState, getVideosFn]);
```

Comment:
As stated in a previous comment, I think you have almost a custom hook, you should be able to have a more agnostic use for these. Let me know if you need help on creating the custom hook. I do think it will simplify the way you fetch for data.